angular/projects/global/src/lib/components/table-template/table-template-utils.ts

```diff
@@ -0,0 +1,87 @@
+import { Injectable } from '@angular/core';
+import { Router, Params } from '@angular/router';
+import { PlatformLocation } from '@angular/common';
+import {
+  TableObject,
+  DEFAULT_TABLE_PAGE_SIZE,
+  DEFAULT_TABLE_CURRENT_PAGE,
+  DEFAULT_TABLE_SORT_BY,
+  DEFAULT_TABLE_KEYWORDS
+} from './table-object';
+
+@Injectable()
+export class TableTemplateUtils {
+  constructor(private platformLocation: PlatformLocation, private router: Router) {}
+
+  public updateUrl(tableObject: TableObject) {
```

Comment:
This function is fairly large, and does a lot of stuff.

It would be my recommendation to break it up into a bunch of small functions, each of which is given just the piece of tableObject they care about, and can handle their own error handling and return values, etc.  And then this function is reduced to just managing currentUrl, and calling all of the other functions, and updating currentUrl based on their responses.  

You then can document each of the smaller functions more easily, without dumping tons of inline comments into this function.  And each of them is now testable in isolation, and your tests will become smaller as a result.   Making updates becomes easier as well, since each function handles all of its own input and output, and doesn't care about what may have come before or after it.